### PR TITLE
Fix assertion failure in circuit.merkle_tree_gadget_weirdness test on Windows

### PR DESCRIPTION
**Pull request by wo01**
zcash-gtest.exe fails
```
[ RUN      ] circuit.merkle_tree_gadget_weirdness
Assertion failed!

Program: C:\zcash-gtest.exe
File: ./snark/libsnark/gadgetlib1/gadgets/basic_gadgets.tcc, Line 50

Expression: this->pb.lc_val(packed).as_bigint().num_bits() <= bits.size()
```

The argument type of `__builtin_clzl()` function is unsigned long.
So, we need to replace `__builtin_clzl` with `__builtin_clzll`. (The argument type of `__builtin_clzll()` is unsigned long long (64bit)).

### CODE REVIEW

File: src/snark/libsnark/algebra/fields/bigint.tcc [hunk 1]
```diff
@@ -124,7 +124,7 @@ size_t bigint<n>::num_bits() const
         }
         else
         {
```
Comment (daira) [inline src/snark/libsnark/algebra/fields/bigint.tcc]: Add ``static_assert(GMP_NUMB_MAX <= ULLONG_MAX, "coercing limb to unsigned long long might truncate");``

**Comment by str4d** [comment]
I removed the test output showing the `circuit.values` failure from the PR description, as it is unrelated to the assertion failure fixed by this PR, and still fails afterwards.

**Comment by wo01** [comment]
zcash-gtest succeeds on Windows and Ubuntu.

**Comment by ebfull** [comment]
ACK

**Comment by mdr0id** [comment]
@charlieok Did you run into this issue while setting up native Windows ?

**Comment by charlieok** [comment]
@mdr0id I probably have. If I understand correctly, it shows up as an "Assertion failed!" message in test output. So far my interest is in making tests run to completion and not necessarily making sure they all pass or reasons why individual tests might have failed.

**Comment by bitcartel** [comment]
@zkbot r+

**Pull request merged**